persist/fs/commitlog/options.go

```diff
@@ -78,6 +82,23 @@ func NewOptions() Options {
 	return o
 }
 
+func defaultRetentionOptions() retention.Options {
+	return retention.NewOptions().
+		SetBlockSize(15 * time.Minute)
+}
+
+func (o *options) Validate() error {
+	if o.FlushInterval() < 0 {
```

Comment:
I tried that originally but reverted due to some tests relying on the ability to stop bg flushing –

```
func TestCommitLogWriteErrorOnFull(t *testing.T) {
	// Set backlog of size one and don't automatically flush
	backlogQueueSize := 1
	flushInterval := time.Duration(0)
	opts, _ := newTestOptions(t, overrides{
		backlogQueueSize: &backlogQueueSize,
		flushInterval:    &flushInterval,
	})
```

If we'd never have bg flushing turned off for production, I'll make the suggested change and re-work the test.